src/values/ObjectValue.js

```diff
@@ -567,14 +567,20 @@ export default class ObjectValue extends ConcreteValue {
       // The original object might be a React props object, thus
       // if it is, we need to ensure we mark it with the same rules
       if (realm.react.enabled && realm.react.reactProps.has(this)) {
+        transferSafePropertiesToRemoveFromObjectsToProps(realm, [this], template);
```

Comment:
It's getting pretty hard for me to understand when we need to call `transferSafePropertiesToRemoveFromObjectsToProps`. How do you know where to put it? What happens if you forget to? Are any tests tracking this?

Reply:
That's a valid point. Let me simplify this.

Reply:
So, I've simplified it in many other cases, but we still need to call it manually in some cases like the one above. I think most cases are now covered though.

Reply:
I did add a comment to the above point though.